mat64/dense.go

```diff
@@ -417,3 +420,25 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	enc := binary.NewEncoder(buf)
+	err := enc.Encode(&m.mat)
```

Comment:
This is not what we want to do.

Consider the case where m is a 10x10 view into a 1000x1000 matrix near the upper left. This would write almost the entire data slice of the 1000x1000, but would not be able to be used to recover the entire 1000x1000 matrix. We need to just write the 100 elements represented by m.